platform/util/platform-util/src/test/java/org/codice/ddf/platform/util/properties/PropertiesLoaderTest.java

```diff
@@ -0,0 +1,200 @@
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PropertiesLoaderTest {
+
+  @Mock private Properties propertiesMock;
+
+  @ClassRule public static TemporaryFolder temporaryFolder = new TemporaryFolder();
+  private static final String PROPERTIES_FILENAME = "test.properties";
+  private static final String NON_EXISTENT_FILENAME = "nonExistentFile.properties";
+  private static File propertiesFile;
+
+  private static Map<Object, Object> exampleMap = ImmutableMap.of("a", "A", "b", "B", "c", "C");
+
+  private static final PropertiesLoader PROPERTIES_LOADER = PropertiesLoader.getInstance();
+
+  @BeforeClass
+  public static void setUpClass() throws Exception {
+
+    // set up string to put in properties file
+    StringBuilder sb = new StringBuilder();
+    for (Map.Entry<Object, Object> entry : exampleMap.entrySet()) {
+      sb.append(entry.getKey() + "=" + entry.getValue() + "\n");
+    }
+
+    // initialize file
+    propertiesFile = temporaryFolder.newFile(PROPERTIES_FILENAME);
+    Files.write(propertiesFile.toPath(), sb.toString().getBytes());
+  }
+
+  @Test
+  public void testToMapWithProperties() throws Exception {
+    when(propertiesMock.entrySet()).thenReturn(exampleMap.entrySet());
+
+    Map<?, ?> entries = PROPERTIES_LOADER.toMap(propertiesMock);
+    Assert.assertEquals("Returns a map of the properties", exampleMap, entries);
+  }
+
+  @Test
+  public void testToMapWithNull() throws Exception {
+    Map<?, ?> entries = PROPERTIES_LOADER.toMap(null);
+    Assert.assertEquals("Returns an empty map", Collections.emptyMap(), entries);
+  }
+
+  @Test
+  public void testToMapWithEmptyProperties() throws Exception {
+    Map<?, ?> entries = PROPERTIES_LOADER.toMap(new Properties());
+    Assert.assertEquals("Returns an empty map", Collections.emptyMap(), entries);
+  }
+
+  @Test
+  public void testLoadPropertiesWithoutPropertiesFile() throws Exception {
+    Properties testProperties = PROPERTIES_LOADER.loadProperties(null, null);
+    Properties emptyProperties = new Properties();
+
+    Assert.assertEquals("Returns empty properties", emptyProperties, testProperties);
+  }
+
+  @Test
+  public void testLoadPropertiesWithEmptyPropertiesFile() throws Exception {
+    File emptyFile = temporaryFolder.newFile("tempFile");
+    Properties testProperties = PROPERTIES_LOADER.loadProperties(emptyFile.getPath(), null);
+    Properties emptyProperties = new Properties();
+
+    Assert.assertEquals("Returns empty properties", emptyProperties, testProperties);
+  }
+
+  @Test
+  public void testLoadPropertiesWithProperties() throws Exception {
+    Properties testProperties = PROPERTIES_LOADER.loadProperties(propertiesFile.getCanonicalPath());
+
+    Assert.assertEquals(
+        "Returns the correct properties", exampleMap.entrySet(), testProperties.entrySet());
```

Comment:
❗️ Since we're testing each strategy individually, this test is pretty worthless / redundant unless we have a deterministic way to verify strategy ordering. Which we should have. 

Another way to put it: This test already tests the functionality tested by `testAttemptLoadWithSpring` because that's the first strategy and it will always succeed. What we need is a way to verify the order of the strategies list without exposing it to the world. That's what this test should truly be testing. Otherwise, just delete it.